Mathlib/Algebra/Group/Fin.lean

```diff
@@ -123,9 +123,55 @@ lemma sub_one_lt_iff {k : Fin (n + 1)} : k - 1 < k ↔ 0 < k :=
   not_iff_not.1 <| by simp only [lt_def, not_lt, val_fin_le, le_sub_one_iff, le_zero_iff]
 #align fin.sub_one_lt_iff Fin.sub_one_lt_iff
 
+lemma lt_sub_iff {n : ℕ} {a b : Fin n} : a < a - b ↔ a < b := by
+  cases' n with n
+  · exact a.elim0
+  by_cases h: a < b
+  · simp only [h, iff_true]
+    rw [lt_iff_val_lt_val, sub_def]
+    simp only
+    obtain ⟨k, hk⟩ := Nat.exists_eq_add_of_lt b.is_lt
+    have : n + 1 - b = k + 1 := by
+      simp_rw [hk]
```

Comment:
Why `simp_rw`?

Reply:
Due to `rw` trying to rewrite inside `b : Fin (n + 1)`, and getting a `motive is not correct` error. Left a comment